kernel/tests/golden_tables.rs

```diff
@@ -0,0 +1,343 @@
+        let mut sorted_expected: Vec<&str> = formatted_expected.trim().lines().collect();
+        let mut sorted_result: Vec<&str> = formatted_result.trim().lines().collect();
+        sort_lines!(sorted_expected);
+        sort_lines!(sorted_result);
+        assert_eq!(
+            sorted_expected, sorted_result,
+            "\n\nexpected:\n\n{:#?}\nactual:\n\n{:#?}\n\n",
+            sorted_expected, sorted_result,
+        );
+    };
+}
+
+// copypasta from DAT
+// some things are equivalent, but don't show up as equivalent for `==`, so we normalize here
+fn normalize_col(col: &Arc<dyn Array>) -> Arc<dyn Array> {
+    if let DataType::Timestamp(unit, Some(zone)) = col.data_type() {
+        if **zone == *"+00:00" {
+            arrow_cast::cast::cast(&col, &DataType::Timestamp(*unit, Some("UTC".into())))
+                .expect("Could not cast to UTC")
+        } else {
+            col.clone()
+        }
+    } else {
+        col.clone()
+    }
+}
+
+// do a full table scan at the latest snapshot of the table and compare with the expected data
+async fn latest_snapshot_test(test_name: &str) -> Result<(), Box<dyn std::error::Error>> {
+    let test_dir = load_test_data(test_name).unwrap();
+    let test_path = test_dir.path().join(test_name);
+    let table_path = std::fs::canonicalize(test_path.join("table")).unwrap();
+    let table_path = url::Url::from_directory_path(table_path).unwrap();
+    let expected_path = std::fs::canonicalize(test_path.join("expected")).unwrap();
+
+    let engine = DefaultEngine::try_new(
+        &table_path,
+        std::iter::empty::<(&str, &str)>(),
+        Arc::new(TokioBackgroundExecutor::new()),
+    )
+    .unwrap();
+
+    let table = Table::new(table_path);
+    let snapshot = table.snapshot(&engine, None).unwrap();
+
+    let scan = snapshot.into_scan_builder().build().unwrap();
+    let scan_res = scan.execute(&engine).unwrap();
+    let batches: Vec<RecordBatch> = scan_res
+        .into_iter()
+        .map(|sr| {
+            let data = sr.raw_data.unwrap();
+            let record_batch = to_arrow(data).unwrap();
+            if let Some(mut mask) = sr.mask {
+                let extra_rows = record_batch.num_rows() - mask.len();
+                if extra_rows > 0 {
+                    // we need to extend the mask here in case it's too short
+                    mask.extend(std::iter::repeat(true).take(extra_rows));
+                }
+                filter_record_batch(&record_batch, &mask.into()).unwrap()
+            } else {
+                record_batch
+            }
+        })
+        .collect();
+
+    let expected = read_expected(&expected_path).await.unwrap().unwrap();
+    let schema: ArrowSchemaRef = Arc::new(scan.schema().as_ref().try_into().unwrap());
+
+    // convert the batch +00:00 to UTC
+    let result: Vec<RecordBatch> = batches
+        .iter()
+        .map(|batch| {
+            let result: Vec<ArrayRef> = batch.columns().iter().map(normalize_col).collect();
```

Comment:
don't think you should need to specify the type here:
```suggestion
            let result = batch.columns().iter().map(normalize_col).collect();
```